questionpy/_attempt.py

```diff
@@ -156,6 +169,10 @@ def __init__(
         self.cache_control = CacheControl.PRIVATE_CACHE
         self.placeholders: dict[str, str] = {}
         self.css_files: list[str] = []
+        self._javascript_calls: dict[JsModuleCall, None] = {}
+        """LMS has to call these JS modules/functions. A dict is used as a set to avoid duplicates and to preserve
+        the insertion order."""
```

Comment:
Warum möchten wir keine Duplikate erlauben?

Reply:
Habe ich in [Tefen](https://tefen.isis.tu-berlin.de/wiki/doku.php?id=questionpy:api:js-api) unter 5. beschrieben.

Reply:
Hmm, damit schränken wir aber allgemein die Funktionalität ein, um ein spezifisches Problem zu lösen. Es gibt sicher auch Funktionen, die mehrmals aufgerufen werden müssen, oder wo bei mehreren Aufrufen der letzte behalten werden sollte und nicht der erste. Jedenfalls ist es für die Paketentwickler*innen so nicht intuitiv.

Vielleicht kannst du eine zweite Funktion `call_js_once` hinzufügen, die keine Duplikate erlaubt, oder einen Parameter `call_only_once`?